### PR TITLE
use remote inspection instead of docker pull

### PR DESCRIPTION
**Pull request by ChaosGramer**
# Remote Manifest Inspection instead of docker pull

We recently updated our AKS to version 1.15.8 and started having problems with this project. We identified, that the new 30 seconds timeout for webhooks introduced in [k8s 1.15](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#mutatingwebhook-v1beta1-admissionregistration-k8s-io) was killing the webhook, because the pull of images could take long time. 
From the Docs:
> 

timeoutSecondsinteger | TimeoutSeconds specifies the timeout for this webhook. After the timeout passes, the webhook call will be ignored or the API call will fail based on the failure policy. The timeout value must be between 1 and 30 seconds. Default to 30 seconds.
-- | --

Therefore we came up with the idea to use remote inspection inspired by [containers/skopeo](https://github.com/containers/skopeo).

Also this should remove the necessity of mounting docker.sock into the container. 

I've tested this version as much as I could. 

Please tell me what you think. 

sidenote: this is my first golang code so please tell me in case there is something weird in this change. 

## Performance 

openjdk:slim has a size of ~ 265MB for us a download took 30 Secs in worst case per image

With the new solution we are down to like 1 Second.

### CODE REVIEW

**Comment by torresdal** [comment]
This is extremely welcome @ChaosGramer ! At the time when we did the implementation it solved a major problem of figuring out the exact command to invoke, but as you point out at the cost of image download time. We looked for a remote/api-solution at the time as well, but couldn't find any. 

I will do a code review shortly. Thanks!

File: cmd/azure-keyvault-secrets-webhook/main.go [hunk 1]
```diff
@@ -191,6 +192,10 @@ func main() {
 		cloudConfigHostPath:  viper.GetString("cloud_config_host_path"),
 	}
 
+	if config.cloudConfigHostPath == "" {
+		config.cloudConfigHostPath = "/etc/kubernetes/azure.json"
+	}
+
```
[the remark this answers is not in the document]
Comment (ChaosGramer) [reply cmd/azure-keyvault-secrets-webhook/main.go]: done

**Comment by torresdal** [comment]
@ChaosGramer merged in version/tag 1.1.0-beta.29 - not fully tested, so going to do that now with this beta release.

Thanks for contributing!

**Comment by torresdal** [comment]
Merged in ff5005876be65fa8e612f6f0cd77326c516f6242

**Pull request closed without merging**